models/data_entities.go

```diff
@@ -2,48 +2,23 @@ package models
-	TransactionID string                       `bson:"transaction_id"`
-	Data          InsolvencyResourceDaoDataDto `bson:"data"`
-}
-
-// InsolvencyResourceDaoDataDto contains the data for the insolvency resource in Mongo with links rather than real data
-type InsolvencyResourceDaoDataDto struct {
-	CompanyNumber      string                         `bson:"company_number"`
-	CaseType           string                         `bson:"case_type"`
-	CompanyName        string                         `bson:"company_name"`
-	Etag               string                         `bson:"etag"`
-	Kind               string                         `bson:"kind"`
-	Practitioners      string                         `bson:"practitioners,omitempty"`
-	Links              InsolvencyResourceLinksDao     `bson:"links,omitempty"`
-	Attachments        []AttachmentResourceDao        `bson:"attachments,omitempty"`
-	Resolution         *ResolutionResourceDao         `bson:"resolution,omitempty"`
-	StatementOfAffairs *StatementOfAffairsResourceDao `bson:"statement-of-affairs,omitempty"`
-	ProgressReport     *ProgressReportResourceDao     `bson:"progress-report,omitempty"`
-}
-
 // InsolvencyResourceDao contains the meta-data for the insolvency resource in Mongo
 type InsolvencyResourceDao struct {
-	ID            primitive.ObjectID         `bson:"_id"`
-	TransactionID string                     `bson:"transaction_id"`
-	Etag          string                     `bson:"etag"`
-	Kind          string                     `bson:"kind"`
-	Data          InsolvencyResourceDaoData  `bson:"data"`
-	Links         InsolvencyResourceLinksDao `bson:"links"`
-}
-
-// InsolvencyResourceDaoData contains the data for the insolvency resource in Mongo
-type InsolvencyResourceDaoData struct {
-	CompanyNumber      string                         `bson:"company_number"`
-	CaseType           string                         `bson:"case_type"`
-	CompanyName        string                         `bson:"company_name"`
-	Practitioners      []PractitionerResourceDao      `bson:"practitioners,omitempty"`
-	Attachments        []AttachmentResourceDao        `bson:"attachments,omitempty"`
-	Resolution         *ResolutionResourceDao         `bson:"resolution,omitempty"`
-	StatementOfAffairs *StatementOfAffairsResourceDao `bson:"statement-of-affairs,omitempty"`
-	ProgressReport     *ProgressReportResourceDao     `bson:"progress-report,omitempty"`
+	ID            primitive.ObjectID `bson:"_id"`
+	TransactionID string             `bson:"transaction_id"`
+	Data          struct {
+		CompanyNumber      string                         `bson:"company_number"`
+		CaseType           string                         `bson:"case_type"`
+		CompanyName        string                         `bson:"company_name"`
+		Etag               string                         `bson:"etag"`
+		Kind               string                         `bson:"kind"`
+		Practitioners      string                         `bson:"practitioners,omitempty"`
```

Comment:
why is the Practitioners data element a string rather than something more structured like Links or Attachments?
That's the requirement now being saved as links in string format.
the comment above this type still refers to 'meta-data'
No idea what you meant????

Reply:
On the first point, I agree that the requirement is to save it as links rather than the inline data, but my question is why it's an unstructured string rather than an array of whatever data type makes sense to hold practitioner links, as this seems to require additional parsing when it is used in onward code.

My second point was that the comment on line 5 used to have the word 'meta-data' in it to distinguish it from the different data type that stored the actual data.  Now that the structures have been merged as part of your simplifying & tidying, this distinction isn't helpful any more so 'meta-data' should be replaced with 'data'.

Reply:
Edit to comment made as requested, thank you.

Discussed adding structure to the practitioner links data & agreed that this could not be done in a similar way to other existing fields since the requirement specifies a map - leaving this as-is for now.